packages/babel-helpers/src/helpers.js

```diff
@@ -1029,3 +1029,34 @@ helpers.classPrivateFieldSet = () => template.program.ast`
     return value;
   }
 `;
+
+helpers.classStaticPrivateFieldLooseBase = () => template.program.ast`
+  export default function _classStaticPrivateFieldLooseBase(receiver, classConstructor) {
+    if (receiver !== classConstructor && receiver.constructor !== classConstructor) {
```

Comment:
What's the `receiver.constructor !== classConstructor` for?

Reply:
Rereading the whole thing, this logic is busted. That's why I want to make sure the tests work as well

Reply:
That's for when the static field gets called from `this`. In that case I'll probably have an instance of the class instead of the constructor itself. I do think that's what we expect from the spec but it's been a while since the last time I dived in the proposal.

Reply:
It has to throw a type error if it's called on an instance of the class. This check should just be the `receiver !== classConstructor`.

Reply:
I'm not sure, I think that if the instance is of the class type, access should work through `this`. I can check that again. However, in the meantime, we can restrict the access so it does not work on the instance while we figure that out.